components/Events/Filters.js

```diff
@@ -0,0 +1,236 @@
+      Saturday: false,
+    },
+    times: {
+      from: '',
+      to: '',
+    },
+    loc: {
+      'Online': false,
+      'In-Person': false,
+    },
+  };
+  const [filterVals, dispatch] = useReducer(reducer, initialState);
+  const [isFilterOpen, setIsFilterOpen] = useState(false);
+  const [searchValue, setSearchValue] = useState('');
+
+  const dateToMinutes = (date) => {
+    const itemMins =
+      date.getHours() * 60 + date.getMinutes();
+    return itemMins;
+  };
+  const filterTime = (filterEvents) => {
+    const isFilterTimeEmpty = Object.keys(filterVals.times).every(
+      (time) => filterVals.times[time] === '',
+    );
+    if (!isFilterTimeEmpty) {
+      const filterFromMins =
+        filterVals.times.from === '' ? 0 : // 12:00 AM if empty
+          parseInt(filterVals.times.from.split(':')[0]) * 60 +
+          parseInt(filterVals.times.from.split(':')[1]);
+      const filterToMins =
+        filterVals.times.to === '' ? 1440 :  // 11:59 PM if empty
+          parseInt(filterVals.times.to.split(':')[0]) * 60 +
+          parseInt(filterVals.times.to.split(':')[1]);
+
+      return filterEvents.filter((item) => {
+        const itemStartMins = dateToMinutes(new Date(item.start));
+        const itemEndMins = dateToMinutes(new Date(item.end));
+        return itemStartMins >= filterFromMins && itemEndMins <= filterToMins;
+      });
+    }
+    return filterEvents;
+  };
+  const filterComm = (filterEvents) => {
+    const isCommEmpty = Object.keys(filterVals.comms).every(
+      (comm) => !filterVals.comms[comm],
+    );
+    if (!isCommEmpty) {
+      return filterEvents.filter((item) => {
+        if (!item.committee) return false;
+        return filterVals.comms[item.committee];
+      });
+    }
+    return filterEvents;
+  };
+  const filterDays = (filterEvents) => {
+    const isDaysEmpty = Object.keys(filterVals.days).every(
+      (day) => !filterVals.days[day],
+    );
+    if (!isDaysEmpty) {
+      return filterEvents.filter((item) => {
+        const date = new Date(item.start);
+        const itemDay = Object.keys(filterVals.days)[date.getDay()];
+        return filterVals.days[itemDay];
+      });
+    }
+    return filterEvents;
+  };
+  const filterLoc = (filterEvents) => {
+    const isLocEmpty = Object.keys(filterVals.loc).every((loc) => !filterVals.loc[loc]);
+    if (!isLocEmpty) {
+      return filterEvents.filter((item) => {
+        const itemLoc = item.location.toLowerCase().includes('zoom') ? 'Online' : 'In-Person';
+        return filterVals.loc[itemLoc];
+      });
+    }
+    return filterEvents;
+  };
+  const handleSearch = () => {
+    // Check search input
+    const lowerCaseSearch = searchValue.toLowerCase();
+    let allEvents = events.filter((item) =>
+      [item.title, item.committee, item.location, item.description].some(
+        (itemsToSearch) =>
+          String(itemsToSearch).toLowerCase().includes(lowerCaseSearch),
+      ),
+    );
+    allEvents = filterTime(allEvents);
```

Comment:
it isn't `allEvents` in that it's not all possible events from the start, probably rename